src/main/java/gov/nasa/pds/objectAccess/ByteWiseFileAccessor.java

```diff
@@ -190,10 +190,21 @@ private void initializeAccessor(URL url, long offset, int length, long records,
       // if for whatever reason we don't read in sufficient bytes
       if (this.totalBytesRead < expectedBytesToRead) {
         if (expectedBytesToRead > fileSizeMinusOffset) {
+          if ((fileSizeMinusOffset / length) * length == fileSizeMinusOffset)
```

Comment:
Doesn't this always evaluate to true? (_a_ ÷ _b_) × _b_ = _a_, right?

Is one of these a floating point number and the other an int?

Reply:
Integer math (its java not python). Since all those variables are ints, (3/2)*2 = 1